lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1191,9 +1192,21 @@ def set_proj_type(self, proj_type, focal_length=None):
                                  f"None for proj_type = {proj_type}")
             self._focal_length = np.inf
 
-    def _roll_to_vertical(self, arr):
-        """Roll arrays to match the different vertical axis."""
-        return np.roll(arr, self._vertical_axis - 2)
+    def _roll_to_vertical(
+        self, arr: "np.typing.ArrayLike", sign: Literal[1, -1] = 1
+    ) -> np.ndarray:
```

Comment:
@QuLogic it seems `mplot3d` is not yet typed at all. Do you plan to go with type stubs or inline type hints here?

Reply:
Correct that all of mpltoolkits (including mplot3d) is not yet type hinted at all.

I have no current plans regarding extending type hints here at the moment (not opposed, just have no active plans to undertake it myself)

I think whether to do stubs or inline is up to whoever wishes to implement it. Stubs would be my default option, just because that is what we have mostly, but it is not a strong opinion.

I _do_ prefer inline in general, but opted for stubs because it limited how much runtime code needed to be touched in an already pretty large task.

That said, we have precedence for some new typing work being done inline (`_mathtext.py`).

Further, I have in general erred towards not type hinting private functionality (as this is), at least in stubs, as they are mostly useful to external users anyway. 


TL;DR Could go either way, not really harming anything to have them here.